src/bloqade/analysis/address/impls.py

```diff
@@ -203,7 +203,10 @@ def apply(
     ):
 
         origin_qubits = tuple(
-            [frame.get(input_elem).origin_qubit for input_elem in stmt.inputs]
+            [
+                frame.get_casted(input_elem, AddressWire).origin_qubit
+                for input_elem in stmt.inputs
+            ]
         )
         new_address_wires = tuple(
             [AddressWire(origin_qubit=origin_qubit) for origin_qubit in origin_qubits]
```

Comment:
This is also wrong you should just return all the values from the frame because there is no dependency on what the value is. 

```
return frame.get_values(stmt.inputs)
```

Reply:
@johnzl-777 how about this one?

Reply:
I talked to @johnzl-777 in person, he is OK with this change.

Reply:
Changed it!